ui/app/registration/controllers/patientCommonController.js

```diff
@@ -17,7 +17,186 @@ angular.module('bahmni.registration')
+                            if ($rootScope.extenstionPatient.id !== $scope.patient.uuid) {
+                                $window.open(Bahmni.Registration.Constants.existingPatient + $rootScope.extenstionPatient.id, "_self");
+                            }
+                        } else $window.open(Bahmni.Registration.Constants.newPatient, "_self");
+                        update($rootScope.extenstionPatient);
+                        $scope.$digest();
+                    }
+                }, false);
+            };
+
+            $scope.isDisabled = function (attribute) {
+                return $scope.attributesToBeDisabled.includes(attribute);
+            };
+
+            function isIdentifierVoided (identifierType) {
+                if ($scope.patient.uuid !== undefined && $rootScope.patientIdentifiers !== null) {
+                    for (var i = 0; i < $rootScope.patientIdentifiers.length; i++) {
+                        var identifier = $rootScope.patientIdentifiers[i];
+                        if (identifier.identifierType.display === identifierType) {
+                            return true;
+                        }
+                    }
+                }
+                return false;
+            }
+
+            $scope.showButton = function (identifierType, identifierValue) {
+                var extenstionPoint = getExtensionPoint(identifierType);
+                if (extenstionPoint != null && identifierValue === undefined) {
+                    if (buttonMap.get(extenstionPoint.id) === identifierType || buttonMap.get(extenstionPoint.id) === undefined) {
+                        if (buttonMap.get(extenstionPoint.id) === undefined) {
+                            buttonMap.set(extenstionPoint.id, identifierType);
+                        }
+                        return !isIdentifierVoided(identifierType);
+                    }
+                }
+                return false;
+            };
+
+            $scope.getDisplayName = function (identifierType) {
+                var extenstionPoint = getExtensionPoint(identifierType);
+                if (extenstionPoint != null) {
+                    return extenstionPoint.extensionParams.linkDisplay;
+                }
+            };
+
+            function getExtensionPoint (identifierType) {
+                if ($scope.regExtPoints !== null) {
+                    for (var i = 0; i < $scope.regExtPoints.length; i++) {
+                        var identifierTypes = $scope.regExtPoints[i].extensionParams.identifierType;
+                        for (var j = 0; j < identifierTypes.length; j++) {
+                            if (identifierType === identifierTypes[j]) {
+                                return $scope.regExtPoints[i];
+                            }
+                        }
+                    }
+                }
+                return null;
+            }
+
+            function update (patient) {
```

Comment:
Can we put better name for this function? if this is about updating information from other sources - say something like updateInfoFromExtSource()

Reply:
updated the function name